httpx/http_service.go

```diff
@@ -113,10 +127,22 @@ func (t *Transport) RoundTrip(ctx context.Context, req *http.Request) (*http.Res
 // retries requests.
 type RetryTransport struct {
 	*retry.Retrier
+	MethodsToRetry []string
 	Transport RoundTripper
 }
 
 func (t *RetryTransport) RoundTrip(ctx context.Context, req *http.Request) (*http.Response, error) {
+	shouldRetry := false
+	for _,method := range t.MethodsToRetry {
```

Comment:
Might be a little simpler to just make this a `map[string]true` (a little faster too, but probably negligible).

```go
if !t.MethodsToRetry[req.Method] {
// ...
}
```